snowflake/src/main/java/org/apache/iceberg/snowflake/JdbcSnowflakeClient.java

```diff
@@ -153,8 +155,9 @@ public boolean databaseExists(SnowflakeIdentifier database) {
                   queryHarness.query(
                       conn, finalQuery, SCHEMA_RESULT_SET_HANDLER, database.databaseName()));
     } catch (SQLException e) {
-      if (e.getErrorCode() == 2003 && e.getMessage().contains("does not exist")) {
-        return false;
+      // 2003, 2043, 2001 are possible Snowflake error codes for Object does not exist
+      if (e.getErrorCode() == 2003 || e.getErrorCode() == 2043 || e.getErrorCode() == 2001) {
+        throw new NoSuchNamespaceException("%s", e.getMessage());
```

Comment:
The semantics of `databaseExists` or `schemaExists` is to return false instead of throwing an exception, so these two places should still just return false.